chips/nrf5x/src/rtc.rs

```diff
@@ -1,28 +1,100 @@
-use core::cell::Cell;
-use core::mem;
+use kernel::common::cells::OptionalCell;
+use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
 use kernel::hil::time::{self, Alarm, Freq32KHz, Time};
 use kernel::hil::Controller;
-use peripheral_registers::{RTC1, RTC1_BASE};
 
-fn rtc1() -> &'static RTC1 {
-    unsafe { mem::transmute(RTC1_BASE as usize) }
+pub const RTC1_BASE: usize = 0x40011000;
+
+#[repr(C)]
+struct RtcRegisters {
+    /// Start RTC Counter.
+    tasks_start: WriteOnly<u32, Task::Register>,
+    /// Stop RTC Counter.
+    tasks_stop: WriteOnly<u32, Task::Register>,
+    /// Clear RTC Counter.
+    tasks_clear: WriteOnly<u32, Task::Register>,
+    /// Set COUNTER to 0xFFFFFFF0.
+    tasks_trigovrflw: WriteOnly<u32, Task::Register>,
+    _reserved0: [u8; 240],
+    /// Event on COUNTER increment.
+    events_tick: ReadWrite<u32, Event::Register>,
+    /// Event on COUNTER overflow.
+    events_ovrflw: ReadWrite<u32, Event::Register>,
+    _reserved1: [u8; 56],
+    /// Compare event on CC[n] match.
+    events_compare: [ReadWrite<u32, Event::Register>; 4],
+    _reserved2: [u8; 436],
+    /// Interrupt enable set register.
+    intenset: ReadWrite<u32, Inte::Register>,
+    /// Interrupt enable clear register.
+    intenclr: ReadWrite<u32, Inte::Register>,
+    _reserved3: [u8; 52],
+    /// Configures event enable routing to PPI for each RTC event.
+    evten: ReadWrite<u32, Inte::Register>,
+    /// Enable events routing to PPI.
+    evtenset: ReadWrite<u32, Inte::Register>,
+    /// Disable events routing to PPI.
+    evtenclr: ReadWrite<u32, Inte::Register>,
+    _reserved4: [u8; 440],
+    /// Current COUNTER value.
+    counter: ReadOnly<u32>,
+    /// 12-bit prescaler for COUNTER frequency (32768/(PRESCALER+1)).
+    /// Must be written when RTC is stopped.
+    prescaler: ReadWrite<u32, Prescaler::Register>,
+    _reserved5: [u8; 52],
+    /// Capture/compare registers.
+    cc: [ReadWrite<u32, CC::Register>; 4],
+    _reserved6: [u8; 2732],
+    /// Peripheral power control.
+    power: ReadWrite<u32>,
 }
 
+register_bitfields![u32,
+    Inte [
+        /// Enable interrupt on TICK event.
+        TICK 0,
+        /// Enable interrupt on OVRFLW event.
+        OVRFLW 1,
+        /// Enable interrupt on COMPARE[0] event.
+        COMPARE0 16,
+        /// Enable interrupt on COMPARE[1] event.
+        COMPARE1 17,
+        /// Enable interrupt on COMPARE[2] event.
+        COMPARE2 18,
+        /// Enable interrupt on COMPARE[3] event.
+        COMPARE3 19
+    ],
+    Prescaler [
+        PRESCALER OFFSET(0) NUMBITS(12)
+    ],
+    Task [
+        ENABLE 0
+    ],
+    Event [
+        READY 0
+    ],
+    CC [
+        CC OFFSET(0) NUMBITS(24)
+    ]
+];
+
 pub struct Rtc {
-    callback: Cell<Option<&'static time::Client>>,
+    regs: *const RtcRegisters,
+    callback: OptionalCell<&'static time::Client>,
 }
 
 pub static mut RTC: Rtc = Rtc {
-    callback: Cell::new(None),
+    regs: RTC1_BASE as *const RtcRegisters,
```

Comment:
Actually, these days we should start using `StaticRef`... as long as we're at it